File: App/src/main/java/com/terminal/ui/TerminalWidget.java
package com.terminal.ui;

import com.terminal.TerminalDisplay;
import com.terminal.TtyConnector;

import javax.swing.*;
import java.awt.*;

/**
 * @author traff
 */
public interface TerminalWidget {
    JediTermWidget createTerminalSession(TtyConnector ttyConnector);

    JComponent getComponent();

    default JComponent getPreferredFocusableComponent() {
        return getComponent();
    }

    boolean canOpenSession();

    void setTerminalPanelListener(TerminalPanelListener terminalPanelListener);

    Dimension getPreferredSize();

    TerminalSession getCurrentSession();

    TerminalDisplay getTerminalDisplay();

    void addListener(TerminalWidgetListener listener);

    void removeListener(TerminalWidgetListener listener);

    void grabFocus();

}

File: App/src/main/java/com/terminal/ui/TerminalWidget.kt
package com.terminal.ui


import com.terminal.TerminalDisplay
import com.terminal.TtyConnector
import java.awt.Dimension
import javax.swing.JComponent

interface TerminalWidget  {
    fun createTerminalSession(ttyConnector: TtyConnector?): CMD3TermWidget
    val component: JComponent

    fun canOpenSession(): Boolean
    fun setTerminalPanelListener(terminalPanelListener: TerminalPanelListener)
    val preferredSize: Dimension
    val currentSession: TerminalSession
    val terminalDisplay: TerminalDisplay
    fun addListener(listener: TerminalWidgetListener)
    fun removeListener(listener: TerminalWidgetListener)
    fun grabFocus()
}